src/index.js

```diff
@@ -1,25 +1,23 @@
-export default function createExtraScopePlugin(...extra) {
-  const scopes = extra.map(scope => `${scope.trim()} `)
-  const seen = new WeakSet()
+export default function createExtraScopePlugin(...scopes) {
+  scopes = scopes.map(scope => `${scope.trim()} `)
 
-  const extraScopePlugin = (
-    context,
-    content,
-    selectors,
-    parents,
-    line,
-    column,
-    length,
-    type,
-  ) => {
-    if (context !== 2 || type === 107 || seen.has(selectors)) return
+  return element => {
+    if (element.type !== 'rule') {
+      return
+    }
 
-    seen.add(selectors)
+    if (element.root?.type === '@keyframes') {
+      return
+    }
 
-    for (let i = 0; i < selectors.length; i++) {
-      selectors[i] = scopes.map(scope => `${scope}${selectors[i]}`).join(',')
+    if (
+      !element.parent ||
```

Comment:
Could you explain the logic behind those conditions here? writing Stylis plugins is tricky and I would like to recheck the thought process behind this - I'm not saying there is anything wrong with it, it would just make it easier for me to review it

Reply:
I'm no longer working on the project that I wrote this for so I can't test out exactly what CSS elements this is targeting, but I do know that it's targeting elements [that pass through the `compat` plugin unchanged](https://github.com/emotion-js/emotion/blob/d1ae87e0f2c40366297d6b9365e8fd91b35e6b97/packages/cache/src/stylis-plugins.js#L80-L81). Without this logic, the extra scope would be applied to nested selectors and pseudo-selectors before being joined to parent selectors by `compat`, resulting in the "joined" selector including repeated references to the extra scope.